tasks/main.yml

```diff
@@ -43,5 +43,15 @@
     loop_var: _host_item
   when: ( hostvars[_host_item].inventory_hostname == inventory_hostname ) and installed_rke2_version.stdout is defined and rke2_version != ( installed_rke2_version.stdout | default({}) )
 
+- name: Restart service when config file is changed
+  ansible.builtin.include_tasks: change_config.yml
+  with_items: "{{ groups[rke2_cluster_group_name] }}"
+  loop_control:
+    loop_var: _host_item
+  when: 
+  - rke2_drain_node_during_upgrade | bool == false
+  - hostvars[_host_item].inventory_hostname == inventory_hostname
+  - config_file_is_changed.changed == true
```

Comment:
Can you please fix the 1st and 3rd condition? 

https://ansible-lint.readthedocs.io/rules/literal-compare/

Reply:
I updated the 1st and 3rd conditions from your guide.

Reply:
Thanks @DungT9x . Why is actually the first condition necessary?

Reply:
Hi @MonolithProjects, I coded the wrong 1st value. It's exactly "when: not rke2_drain_node_during_upgrade". I added this condition because when it will be upgraded the system, it was included a restart service task.
Thanks for your question.

Reply:
ok thanks for info